src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/DefaultTagHelperTooltipFactory.cs

```diff
@@ -381,5 +403,252 @@ private MarkupKind GetMarkupKind()
                 return MarkupKind.PlainText;
             }
         }
+
+        public override bool TryCreateTooltip(AggregateBoundElementDescription elementDescriptionInfo, out RazorClassifiedTextElement tagHelperDescription)
+        {
+            var associatedTagHelperInfos = elementDescriptionInfo.AssociatedTagHelperDescriptions;
+            if (associatedTagHelperInfos.Count == 0)
+            {
+                tagHelperDescription = null;
+                return false;
+            }
+
+            // Generates a ClassifiedTextElement that looks something like:
+            //     Namespace.TypeName
+            //     Summary description
+            // with the specific element parts classified appropriately.
+            // Additional entries are separated by two new lines.
+
+            var runs = new List<RazorClassifiedTextRun>();
+            foreach (var descriptionInfo in associatedTagHelperInfos)
+            {
+                if (runs.Count > 0)
+                {
+                    runs.Add(NewLine);
+                    runs.Add(NewLine);
+                }
+
+                ClassifyTypeName(runs, descriptionInfo.TagHelperTypeName);
```

Comment:
Hmmm, are we able to add a corresponding icon in the beginning? Aka like:
![image](https://user-images.githubusercontent.com/2008729/116133583-bc006080-a683-11eb-9ca0-4f206c9a7872.png)

Ditto for the attribute case